### PR TITLE
Load balancer

### PR DESCRIPTION
**Pull request by haim-kermany**
this is basic implementation of the load balancer.
1. @zivnevo , please review the drawio package. for now, I did as minimal changes as needed, for easy code review.
 later there is need to make some code reformating, because there is some duplication of code
 2. @adisos, there are still a lot of "todo" in the code. however, I think I need a feedback at this stage. to see that I am in the right direction. 
 currently, the parsing of the load balancer is not called.

### CODE REVIEW

File: pkg/ibmvpc/ibmDrawioGenerator.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ func (v *Vpe) ShowOnSubnetMode() bool                  { return false }
 func (pgw *PublicGateway) ShowOnSubnetMode() bool      { return true }
 func (fip *FloatingIP) ShowOnSubnetMode() bool         { return false }
 func (tgw *TransitGateway) ShowOnSubnetMode() bool     { return true }
+func (lb *LoadBalancer) ShowOnSubnetMode() bool        { return false }
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/ibmvpc/ibmDrawioGenerator.go:22]: will implement later
https://github.com/np-guard/vpc-network-config-analyzer/issues/260

File: pkg/drawio/iconTreeNode.go [hunk 2]
```diff
@@ -219,6 +223,48 @@ func newVpeTreeNode(parent SquareTreeNodeInterface, name string, resIPs []TreeNo
...
+// ///////////////////////////////////////////
+type LoadBalancerTreeNode struct {
+	abstractIconTreeNode
+	PrivateIPs []TreeNodeInterface
+}
+
+func GroupPrivateIPsWithLoadBalancer(parent SquareTreeNodeInterface, name string, privateIPs []TreeNodeInterface) {
+	switch {
+	case len(privateIPs) == 1:
+		privateIPs[0].(*PrivateIPTreeNode).setLoadBalancer(name)
```
[the remark this answers is not in the document]
Comment (zivnevo) [reply pkg/drawio/iconTreeNode.go:235]: @kyorav , does this make sense?
Suppose a load-balancer has just one private IP, should we put the LB icon inside the subnet?

File: pkg/ibmvpc/vpc.go [hunk 3]
```diff
@@ -191,6 +202,35 @@ func (v *Vpe) Zone() (*Zone, error) {
...
+	listeners []LoadBalancerListener
+}
+
+func (lb *LoadBalancer) Nodes() []vpcmodel.Node {
+	return lb.nodes
+}
+func (lb *LoadBalancer) AddressRange() *ipblock.IPBlock {
+	return nodesAddressRange(lb.nodes)
+}
+func (lb *LoadBalancer) NLis() int {
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/ibmvpc/vpc.go:225]: done

File: pkg/vpcmodel/abstractVPC.go [hunk 4]
```diff
@@ -174,6 +174,9 @@ type Subnet interface {
 	CIDR() string
 }
 
+// todo: elaborate - get list of servers, expandability,...
+type LoadBalancer NodeSet
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/vpcmodel/abstractVPC.go:178]: done

File: pkg/drawio/iconTreeNode.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyorav) [reply pkg/drawio/iconTreeNode.go:235]: An ALB will always have exactly two private IPs allocated to it. A single reserved IP can only happen as a transient state when there is a zonal catastrophic failure and the LB is in the process of allocating a new reserved IP. I think we can ignore this situation.

What might be possible is that both private IPs are in the same subnet. I think we should still draw the ALB outside of the subnet.

Note that the rules for NLB may be a little different, but I haven't researched it thoroughly yet so I don't want to decide on it ATM.

[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/drawio/iconTreeNode.go:235]: this is my suggestion, I whoed it to Karen and she said its OK.  we can discuss with karen more

BTW, it is more complicated than that:
the user gives a list of subnets (might be more than two).
two subnets are chosen arbitrary, and the private IPs are being put at these two subnets.
if the user gives only subnet, there are two IPs, that behave the same, at the same subnet.
image

[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/drawio/iconTreeNode.go:235]: so, in case of two pips in the same subnet, should I draw two pip icons?

**Review by zivnevo — COMMENTED** [review]
LGTM

File: pkg/drawio/iconTreeNode.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/drawio/iconTreeNode.go]: implementation as at the discution

**Review by adisos — APPROVED** [review]
looks good

**Pull request merged**